blog/2024-02-25-import_executor.md

```diff
@@ -0,0 +1,18 @@
+---
+author: J. Nick Koston
+authorURL: https://github.com/bdraco
+title: Importing integrations that may block the event loop
+---
+
+Starting from Home Assistant 2024.3, `import_executor` can be specified in the `manifest.json` to facilitate importing an integration in the executor to avoid blocking the event loop. Setting this flag can significantly improve startup stability for integrations that do significant work at import time.
+
+Debug logging is available by enabling the `homeassistant.loader` logger to help integration code owners determine import times. Consider enabling this flag if warnings are observed when using [asyncio debug mode](https://docs.python.org/3/library/asyncio-dev.html#asyncio-debug-mode) or import times take more than a few hundred milliseconds.
```

Comment:
On what device should we test this? I can imagine a desktop processor would have less performance issues than a RPi3 or so.

Reply:
We should test it on the worst supported hardware